Uplift/Views/DropdownView.swift

```diff
@@ -0,0 +1,201 @@
+//
+//  DropdownView.swift
+//  
+//
+//  Created by Cameron Hamidi on 10/7/19.
+//
+
+import Foundation
+import UIKit
+import SnapKit
+
+enum DropdownStatus {
+    case closed, half, open
+}
+
+enum DropdownDirection {
+    case horizontal, vertical
+}
+
+protocol DropdownViewDelegate: class {
+    func dropdownViewClosed(sender dropdownView: DropdownView)
+    func dropdownViewOpen(sender dropdownView: DropdownView)
+    func dropdownViewHalf(sender dropdownView: DropdownView)
+}
+
+class DropdownView: UIView {
+    
+    // MARK: - INITIALIZATION
+    private var contentView: UIView!
+    private var contentViewHeight: CGFloat!
+    private weak var delegate: DropdownViewDelegate!
+    private var halfCloseDropdownGesture: UITapGestureRecognizer!
+    private var halfCloseView: UIView?
```

Comment:
can we do `private let halfCloseView = UIView()` if possible. same comment for the rest of the views

Reply:
As with `contentView`, those other `UIView`s are supplied as initialization arguments